plugins/builds/index.js

```diff
@@ -203,22 +223,55 @@ exports.register = (server, options, next) => {
                     jobName: nextJobName,
                     username,
                     scmContext,
-                    build
+                    build // this is the parentBuild for the next build
                 };
 
                 // Just start the build if falls in to these 2 scenarios
                 // 1. No join
                 // 2. ([~D,B,C]->A) currentJob=D, nextJob=A, joinList(A)=[B,C]
                 //    joinList doesn't include C, so start A
                 if (joinList.length === 0 || !joinListNames.includes(currentJobName)) {
-                    return startBuild(buildConfig);
+                    return createBuild(buildConfig);
                 }
 
-                // If no parent event id, start if all jobs in the list are done
                 if (!event.parentEventId) {
+                    let finishedBuilds = [];
```

Comment:
We also need this logic for restarting jobs from a past event. Should probably move logic down to after https://github.com/screwdriver-cd/screwdriver/pull/921/files#diff-7a514e440aa937beee722c7ac84dfb93R289